src/main/scala/org/hammerlab/guacamole/commands/jointcaller/evidence/MultiSampleSingleAlleleEvidence.scala

```diff
@@ -97,46 +125,62 @@ case class MultiSampleSingleAlleleEvidence(parameters: Parameters,
   /** Negative log10 prior probability for the given mixture in RNA. */
   private def somaticPriorRna(mixture: Map[String, Double]): Double = {
     val contents = mixture.filter(_._2 > 0).keys.toSet
-    if (contents == Set(allele.ref)) {
+    if (contents == Set(allele.ref))
       0
-    } else if (contents == Set(allele.ref, allele.alt)) {
+    else if (contents == Set(allele.ref, allele.alt))
       parameters.somaticNegativeLog10VariantPriorRna
-    } else {
+    else
       Double.MaxValue
-    }
   }
 
   /** Log10 posterior probabilities for a somatic variant in each tumor RNA sample.  */
   val perTumorRnaSampleSomaticPosteriors: Map[SampleId, Map[AlleleMixture, Double]] = {
-    inputs.items.filter(_.tumorRNA).map(input => {
-      val likelihoods = allEvidences(input.index).asInstanceOf[TumorRNASingleSampleSingleAlleleEvidence].logLikelihoods
-      input.index -> likelihoods.map(kv => (kv._1 -> (kv._2 - somaticPriorRna(kv._1))))
-    }).toMap
+    inputs
+      .items
+      .filter(_.tumorRNA)
+      .map(
+        input => {
+
+          val likelihoods =
+            allEvidences(input.index)
+              .asInstanceOf[TumorRNASingleSampleSingleAlleleEvidence]
+              .logLikelihoods
+
+          input.index ->
+            (
+              for {
+                (alleles, likelihood) <- likelihoods
+              } yield
+                alleles -> (likelihood - somaticPriorRna(alleles))
+            )
+        }
+      ).toMap
   }
 
   /** Maximum a posteriori somatic mixtures for each tumor sample. */
   val perTumorRnaSampleTopMixtures = perTumorRnaSampleSomaticPosteriors.mapValues(_.maxBy(_._2)._1)
 
   /** Indices of tumor rna samples with expression */
-  val tumorRnaSampleExpressed: PerSample[SampleId] = perTumorRnaSampleTopMixtures
-    .filter(pair => pair._2.keys.toSet != Set(germlineAlleles._1, germlineAlleles._2))
-    .keys.toVector
+  val tumorRnaSampleExpressed: PerSample[SampleId] =
+    perTumorRnaSampleTopMixtures
+      .filter(pair => pair._2.keys.toSet != Set(germlineAlleles._1, germlineAlleles._2))
```

Comment:
Seems like a reasonable replacement to me. it seems like the right density to me. Having the `.filter` split up onto multiple lines would be too spacey, so I'm glad this replacement does not do that